lib/src/specs/expression.dart

```diff
@@ -39,15 +39,89 @@ abstract class Expression implements Spec {
     return new BinaryExpression._(toExpression(), other, '&&');
   }
 
+  /// This expression preceding by `await`.
+  Expression get awaited {
+    return new BinaryExpression._(
+      const LiteralExpression._('await'),
+      this,
+      '',
+    );
+  }
+
+  /// Return `{other} = {this}`.
+  Expression assign(Expression other) {
```

Comment:
Hmm, `a.assign(b)` reads way too much like `a = b`. Maybe flip the logic and make it `{this} = {other}`?

Or perhaps rename to `assignTo`? Although not sure that helps much.